### ISSUE TITLE
Can we limit the number of search result?

### ISSUE THREAD

**Issue by zfy0701** [vakata/jstree#1100]
I have a huge jstree and some times searching with one or two character would result huge number of results that frozen the UI, I wonder if we could limit the number of search results to improve performance.

**Comment by vakata** [comment]
As you may notice - there is no jsTree functionality that calls `search()` - it is a function that you call from your code (possibly using an input field, etc), If you want to limit the char count to some minimum amount (or do any custom logic for that matter), you have to do it in your code - it is much easier than providing countless config options in jstree. 

Best regards,
Ivan

**Comment by zfy0701** [comment]
I understood, but limit the char count of input is nearly not enough, there are logs of cases that even you enter 5 or characters the results are still too much, while in some cases one character is perfectly fine.

I would be happy to contribute patches if this feature is blessed and would be great if you could give us some guidance how to do this.

**Comment by vakata** [comment]
As it is you have full manual control over what is considered a match using the `search_callback` function, I believe that is more than enough to achieve whatever you need.
I do not want to pollute the plugin with countless options that are very rarely used and can be achieved with a few lines of code using one of the existing options.
I also do not see how a large number of matches can freeze the UI unless you are using `show_only_matches` which I believe has a note in the docs about being slow in large trees and to use at your own risk.

So to sum up - I believe you can achieve a limit on the search results using `search_callback`: http://www.jstree.com/api/#/?q=search_call&f=$.jstree.defaults.search.search_callback With the easiest solution being a counter which once hit will cause the function to only return false.

Best regards,
Ivan
